folly/CPortability.h

```diff
@@ -122,10 +122,17 @@
 
 #if FOLLY_SANITIZE
 /**
- * GCC 8.2.0 and Clang 6.0 both support the no_sanitize attribute, however,
- * GCC 7.3.0 does not. Check for alternative attributes if no_sanitize is not
- * available.
+ * GCC 8.2.0 asupports the no_sanitize attribute, however, GCC 7.3.0 does not.
+ * Check for alternative attributes if no_sanitize is not available. For Clang,
+ * we should use the __no_sanitize__ attribute.
  */
+#if defined(__clang__)
```

Comment:
This seems a bit brittle, esp given you're not checking clang versions and maybe in future they change again.

Can you not structure this so you check for each attribute in turn regardless of GCC or clang, and pick the "best" one available?

Reply:
Should be less brittle now. They use __has_attribute without checking if it's defined above (at least under clang) so I've done the same also. Using __has_attribute instead of __has_cpp_attribute makes things simpler too